firstvoices/backend/search/utils/query_builder_utils.py

```diff
@@ -3,17 +3,42 @@
 from backend.search.indices.dictionary_entry_document import (
     ELASTICSEARCH_DICTIONARY_ENTRY_INDEX,
 )
+from backend.search.utils.constants import VALID_DOCUMENT_TYPES
 
 
-def get_indices():
+def get_valid_document_types(input_types):
+    allowed_values = VALID_DOCUMENT_TYPES
+
+    if not input_types:
+        return allowed_values
+
+    values = input_types.split(",")
+    selected_values = [
+        value.strip().lower()
+        for value in values
+        if value.strip().lower() in allowed_values
+    ]
+
+    if len(selected_values) == 0:
+        return None
+
+    return selected_values
+
+
+def get_indices(types):
     """
     Returns list of indices to go through depending on the docType
     WORD|PHRASE = ELASTICSEARCH_DICTIONARY_ENTRY_INDEX
     SONG = ELASTICSEARCH_SONG_INDEX
     STORY = ELASTICSEARCH_STORY_INDEX
     """
-    list_of_indices = [ELASTICSEARCH_DICTIONARY_ENTRY_INDEX]
-    return list_of_indices
+    indices = set()
+
+    for doc_type in types:
+        if doc_type == "WORD" or doc_type == "PHRASE":
+            indices.add(ELASTICSEARCH_DICTIONARY_ENTRY_INDEX)
```

Comment:
Extremely minor: "doc_types" are from nuxeo. These can just be types

Reply:
Yes, that was my first attempt as well haha. So the "type" variables gives a warning prompt of it shadowing a internal variable. To avoid that, I went with this name. But I can always rename it if required. This name is only used here and then to test this functionality.